datasource 세션 연결 불가

Caused by: jakarta.persistence.PersistenceException: [PersistenceUnit: metrics] Unable to build Hibernate SessionFactory; nested exception is java.lang.IllegalArgumentException: dataSource or dataSourceClassName or jdbcUrl is required.

--- a/src/main/java/check/demo/config/MetricsDbConfig.java
+++ b/src/main/java/check/demo/config/MetricsDbConfig.java
@@ -4,10 +4,10 @@ import jakarta.persistence.EntityManagerFactory;
 import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.boot.context.properties.ConfigurationProperties;
 import org.springframework.boot.jdbc.DataSourceBuilder;
+import org.springframework.boot.orm.jpa.EntityManagerFactoryBuilder;
 import org.springframework.context.annotation.*;
 import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
 import org.springframework.orm.jpa.*;
-import org.springframework.orm.jpa.vendor.HibernateJpaVendorAdapter;
 import org.springframework.transaction.PlatformTransactionManager;
 
 import javax.sql.DataSource;
@@ -30,22 +30,20 @@ public class MetricsDbConfig {
 
     @Primary
     @Bean(name = "metricsEmf")
-    public LocalContainerEntityManagerFactoryBean metricsEmf() {
-        var vendor = new HibernateJpaVendorAdapter();
-        var emf = new LocalContainerEntityManagerFactoryBean();
-        emf.setDataSource(metricsDataSource());
-        emf.setPackagesToScan("check.demo.model.metrics"); // HealthMetric 엔티티 패키지
-        emf.setJpaVendorAdapter(vendor);
-        emf.setPersistenceUnitName("metrics");
-        emf.setJpaPropertyMap(Map.of(
-                "hibernate.hbm2ddl.auto", "validate",
-                "hibernate.dialect", "org.hibernate.dialect.MariaDBDialect",
-                "hibernate.show_sql", "false"
-        ));
-        return emf;
+    public LocalContainerEntityManagerFactoryBean metricsEmf(EntityManagerFactoryBuilder builder,
+                                                             @Qualifier("metricsDataSource") DataSource dataSource) {
+        return builder
+                .dataSource(dataSource)
+                .packages("check.demo.model.metrics")
+                .persistenceUnit("metrics")
+                .properties(Map.of(
+                        "hibernate.hbm2ddl.auto", "validate",
+                        "hibernate.dialect", "org.hibernate.dialect.MariaDBDialect",
+                        "hibernate.show_sql", "true"
+                ))
+                .build();
     }
 
-    @Primary
     @Bean(name = "metricsTx")
     public PlatformTransactionManager metricsTx(@Qualifier("metricsEmf") EntityManagerFactory emf) {
         return new JpaTransactionManager(emf);

--- a/src/main/java/check/demo/config/ReadDbConfig.java
+++ b/src/main/java/check/demo/config/ReadDbConfig.java
@@ -4,10 +4,10 @@ import jakarta.persistence.EntityManagerFactory;
 import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.boot.context.properties.ConfigurationProperties;
 import org.springframework.boot.jdbc.DataSourceBuilder;
+import org.springframework.boot.orm.jpa.EntityManagerFactoryBuilder;
 import org.springframework.context.annotation.*;
 import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
 import org.springframework.orm.jpa.*;
-import org.springframework.orm.jpa.vendor.HibernateJpaVendorAdapter;
 import org.springframework.transaction.PlatformTransactionManager;
 
 import javax.sql.DataSource;
@@ -28,19 +28,20 @@ public class ReadDbConfig {
     }
 
     @Bean(name = "readEmf")
-    public LocalContainerEntityManagerFactoryBean readEmf() {
-        var vendor = new HibernateJpaVendorAdapter();
-        var emf = new LocalContainerEntityManagerFactoryBean();
-        emf.setDataSource(readDataSource());
-        emf.setPackagesToScan("check.demo.model.read"); // Cctv 엔티티 패키지
-        emf.setJpaVendorAdapter(vendor);
-        emf.setPersistenceUnitName("read");
-        emf.setJpaPropertyMap(Map.of(
-                "hibernate.hbm2ddl.auto", "validate",
-                "hibernate.dialect", "org.hibernate.dialect.MariaDBDialect",
-                "hibernate.show_sql", "false"
-        ));
-        return emf;
+    public LocalContainerEntityManagerFactoryBean readEmf(
+            EntityManagerFactoryBuilder builder,
+            @Qualifier("readDataSource") DataSource dataSource) {
+
+        return builder
+                .dataSource(dataSource)
+                .packages("check.demo.model.read")  // Cctv 엔티티 패키지
+                .persistenceUnit("read")
+                .properties(Map.of(
+                        "hibernate.hbm2ddl.auto", "validate",
+                        "hibernate.dialect", "org.hibernate.dialect.MariaDBDialect",
+                        "hibernate.show_sql", "false"
+                ))
+                .build();
     }
 
     @Bean(name = "readTx")

--- a/src/main/java/check/demo/model/read/Cctv.java
+++ b/src/main/java/check/demo/model/read/Cctv.java
@@ -5,12 +5,13 @@ import lombok.Getter;
 import lombok.Setter;
 
 @Entity
-@Table(name = "cctv_read")
+@Table(name = "cctv")
 @Getter @Setter
 public class Cctv {
     @Id
     private Long id;
 
-    @Column(name = "ip_address", nullable = false)
+//    @Column(name = "ip_address", nullable = false)
+    @Column(name = "ip_address")
     private String ipAddress;
 }

--- a/src/main/java/check/demo/scheduler/HealthCheckScheduler.java
+++ b/src/main/java/check/demo/scheduler/HealthCheckScheduler.java
@@ -4,6 +4,7 @@ import check.demo.model.read.Cctv;
 import check.demo.repository.read.CctvRepository;
 import check.demo.service.HealthCheckService;
 import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
 import org.springframework.scheduling.annotation.Scheduled;
 import org.springframework.stereotype.Component;
 import org.springframework.transaction.annotation.Transactional;
@@ -12,6 +13,7 @@ import java.util.List;
 
 @Component
 @RequiredArgsConstructor
+@Slf4j
 public class HealthCheckScheduler {
 
     private final HealthCheckService service;
@@ -22,6 +24,12 @@ public class HealthCheckScheduler {
     @Scheduled(cron = "*/10 * * * * *")
     public void run() {
         List<Cctv> targets = cctvRepository.findAll();
+        // 한 줄씩 로그 출력
+        for (Cctv cctv : targets ) {
+            log.info("CCTV Loaded -> id={}, ip={}",
+                    cctv.getId(),
+                    cctv.getIpAddress());
+        }
         for (Cctv t : targets) {
             if (t.getIpAddress() != null && !t.getIpAddress().isBlank()) {
                 service.check(t.getId(), t.getIpAddress()); // ICMP + ffprobe

--- a/src/main/resources/.env.example
+++ b/src/main/resources/.env.example
@@ -3,7 +3,12 @@ RTSP_PASSWORD=your_rtsp_password
 RTSP_PORT=554
 RTSP_PATH=your_rtsp_path
 
-DB_DRIVER_CLASS_NAME=your_Driver_class_name
-DB_URL=your_database_url
-DB_USERNAME=your_username
-DB_PASSWORD=your_password
+METRICS_DB_DRIVER_CLASS_NAME=your_Driver_class_name
+METRICS_DB_URL=your_database_url
+METRICS_DB_USERNAME=your_username
+METRICS_DB_PASSWORD=your_password
+
+READ_DB_DRIVER_CLASS_NAME=your_Driver_class_name
+READ_DB_URL=your_database_url
+READ_DB_USERNAME=your_username
+READ_DB_PASSWORD=your_password

--- a/src/main/resources/application.properties
+++ b/src/main/resources/application.properties
@@ -4,15 +4,15 @@ RTSP_PASSWORD=${RTSP_PASSWORD}
 RTSP_PORT=${RTSP_PORT}
 RTSP_PATH=${RTSP_PATH}
 
-# ==== READ DB (cctv_read; 172.30.29.106) ====
+# ==== READ DB (cctv_read; 172.30.29.236) ====
 spring.datasource.read.driver-class-name=${READ_DB_DRIVER_CLASS_NAME}
-spring.datasource.read.url=${READ_DB_URL}
+spring.datasource.read.jdbc-url=${READ_DB_URL}
 spring.datasource.read.username=${READ_DB_USERNAME}
 spring.datasource.read.password=${READ_DB_PASSWORD}
 
-# ==== METRICS DB (health_metrics; 172.30.29.107) ====
+# ==== METRICS DB (health_metrics; 172.30.29.237) ====
 spring.datasource.metrics.driver-class-name=${METRICS_DB_DRIVER_CLASS_NAME}
-spring.datasource.metrics.url=${METRICS_DB_URL}
+spring.datasource.metrics.jdbc-url=${METRICS_DB_URL}
 spring.datasource.metrics.username=${METRICS_DB_USERNAME}
 spring.datasource.metrics.password=${METRICS_DB_PASSWORD}
 

